### PR TITLE
Fixes transpose bug in dataloader

### PR DESCRIPTION
**Pull request by RudrakshTuwani**
The current implementation of transpose assumes that there is only one feature array. In many situations, the DataSet object returns multiple features, in which case the first feature ends up being transposed whereas the rest is left as it is. This update makes the transpose implementation generic and lets it handle multiple features.

### CODE REVIEW

**Comment by jph00** [comment]
I'm not sure this solution is quite right - seems that it will create a list of indep vars, even if there's just one. And if the indep vars is a tensor, it'll turn it into a list.

Might be a good idea to write a few tests of various edge cases as part of fixing this PR...

File: fastai/dataloader.py [hunk 1]
```diff
@@ -73,8 +73,8 @@ def np_collate(self, batch):
 
     def get_batch(self, indices):
         res = self.np_collate([self.dataset[i] for i in indices])
```
Comment (RudrakshTuwani) [inline fastai/dataloader.py]: res is itself a list of length equal to the number of outputs of __getitem__ method of DataSet. The old dataloader assumed this length to be 2 and transposed **only** the first array if transpose was set to True. The update makes the assumption that the last array in res list contains our target label, whereas all the preceding arrays are features.  It thus, contrary to the old dl, transposes all the arrays up to the last array in the res list if transpose is set to True.

Comment (jph00) [reply fastai/dataloader.py]: Thanks - so it works fine with DataLoader - but looks like it is indeed listifying the inputs, even if they're not originally lists. That isn't the expected behaviour of a DataSet, and might break some code.

Comment (jph00) [reply fastai/dataloader.py]: DataSets are not only used with DataLoader. They're also used directly. If you have just a single tensor as x, you're returning a list [x], rather than leaving it as-is. Does that make sense?

Comment (RudrakshTuwani) [reply fastai/dataloader.py]: Oh, I haven't made any changes to the DataSets, just the get_batch method of DataLoaders in case of transpose being set True. Also, the new dataloader doesn't return a list.
```
*x,y = next(iter(trn_dl))
```
The asterisk (*) is actually the reason why x ends up becoming a list. If you instead use the following, then each xi is a Tensor.
```
x1,....,xn, y = next(iter(trn_dl)) 
```
